### DISCUSSION TITLE
cy.intercept catching urls its not supposed to

### DISCUSSION THREAD

**Discussion by CinkadeusBG** [cypress-io/cypress#14900] [Questions and Help]
We're upgrading from v5 to v6 and replacing our cy.route instances in all our test... The problem is the matching is not working the same as it did with cy.route and its catching URLs we don't intend it to.. Its making conversions of the following code difficult.

`cy.intercept("POST", "/Api/Companies/1/Surety", {}).as("CreateSuretyCompany");`
`cy.intercept("POST", "/Api/Companies/1/Surety/*/Map", {}).as("MapSuretyCompany");`

Whenever the /Map url is called, the intercept above it is the one being fired.  What do we need to do to have the intercept work more like cy.route did in regards to url matching?

**Comment by jennifer-shehane** [comment]
`cy.intercept()` is not exactly the same as `cy.route()`. The path matching is too broad for `/Api/Companies/1/Surety`, you need to it be more specific so that all routes with path `/Api/Companies/1/Surety` and `/Api/Companies/1/Surety/*/Map` don't both match.
